cmake/modules/platform/toolcfg/xlc.cmake

```diff
@@ -103,19 +103,21 @@ elseif(OMR_OS_ZOS)
 		"\"-Wc,enum(4)\""              # Specifies how many bytes of storage enums occupy
 		"\"-Wa,goff\""                 # Assemble into GOFF object files
 		"\"-Wc,NOANSIALIAS\""          # Do not generate ALIAS binder control statements
-		"\"-Wc,TARGET(zOSV1R13)\""     # Generate code for the target operating system
+		"\"-Wc,TARGET(${OMR_ZOS_COMPILE_TARGET})\""     # Generate code for the target operating system
 	)
 
 	list(APPEND OMR_PLATFORM_C_COMPILE_OPTIONS
-		"\"-Wc,ARCH(7)\""
-		"\"-Wc,TUNE(10)\""
+		"\"-Wc,ARCH(${OMR_ZOS_COMPILE_ARCHITECTURE})\""
+		"\"-Wc,TUNE(${OMR_ZOS_COMPILE_TUNE})\""
+		"\"-Wl,compat=${OMR_ZOS_LINK_COMPAT}\""
```

Comment:
Why is `-Wl,compat` added here (and to `OMR_PLATFORM_CXX_COMPILE_OPTIONS` below)? It is a link option, not a compile option.

Reply:
This is the way the non-cmake build works. I added it because it was missing for cmake.
https://github.com/eclipse/omr/blob/master/omrmakefiles/rules.zos.mk#L63

Reply:
See also https://github.com/eclipse/omr/blob/master/omrmakefiles/rules.zos.mk#L57

Reply:
The prefix `-Wl` means the option is for the linker. I suggest it shouldn't be in compile flags and the comment in `rules.zos.mk` seems to agree:
```
    # OMRTODO: The COMPAT=ZOSV1R13 option does not appear to be related to
    # optimizations.  This linker option is supplied only on the compile line,
```

Reply:
I think the relevant option is `-Wc,TARGET(zOSV1R13)` which informs the compiler that it's safe to generate object files containing 'conditional sequential RLDs' (see [COMPAT: Binder level option](https://www.ibm.com/support/knowledgecenter/SSLTBW_2.1.0/com.ibm.zos.v2r1.ieab100/compat.htm)).

Reply:
What do you think is happening with the `-Wl,compat=ZOSV1R13` option, it's just being ignored? I expect we should put the option on the linker command lines, I'm not comfortable removing it completely.

Reply:
I can leave the cmake change for `-Wl,compat=` or remove it. cmake isn't being used for zos anyway. I'm hoping this can be merged, and then further improvement related to `-Wl,compat=` can occur in another PR.

Reply:
I'd like to see `-Wl,compat` removed, but I'm fine with deferring that to a separate PR.